azure/services/virtualnetworks/virtualnetworks.go

```diff
@@ -150,19 +154,29 @@ func (s *Service) IsManaged(ctx context.Context) (bool, error) {
 	ctx, _, done := tele.StartSpanWithLogger(ctx, "virtualnetworks.Service.IsManaged")
 	defer done()
 
+	// Report errors if the client is throttled.
+	if s.RetryAfter.Reader.After(time.Now()) {
+		return false, errors.Errorf("Retry-After has not yet expired")
+	}
 	spec := s.Scope.VNetSpec()
 	if spec == nil {
 		return false, errors.New("cannot get vnet to check if it is managed: spec is nil")
 	}
 
 	vnetIface, err := s.Get(ctx, spec)
-	if err != nil {
-		return false, err
-	}
 	vnet, ok := vnetIface.(network.VirtualNetwork)
 	if !ok {
 		return false, errors.Errorf("%T is not a network.VirtualNetwork", vnetIface)
 	}
+	if err != nil {
```

Comment:
Doesn't this error check need to happen before it tries to cast the type into the vnet variable in case it didn't get a response due to rate limiting?

Reply:
no longer an open question as 429 foo is gone!